src/notebook.h

```diff
@@ -424,6 +424,12 @@ class MKCAL_EXPORT Notebook
     */
     QList<QByteArray> customPropertyKeys() const;
 
+    /**
+       Provide the email address associated to the notebook, if any.
+       The address is retrieved via the plugin mechanism, see setPluginName().
+    */
+    QString emailAddress() const;
```

Comment:
Why do we need or want this here when it's already and still available on ServiceHandler? The Notebook class was earlier just a data container without references to other classes while this adds references outside of it. Also there's account on the Notebook too already.

Reply:
I'm proposing this extension, because this `emailAddress` case is like a property of a Notebook. Accounts have various way to set-up the notebook email address, and thus provide via the ServiceHandler plugin a way to get this property.

For me, it looks more intentional to write `email = notebook->emailAdress();` than `email = mKCal::ServiceHandler::instance()->emailAdress(notebook);`. `sendResponse()` for instance is more like an action that is handled by a plugin, and calling it with `mKCal::ServiceHandler::instance()->sendResponse()` seems logical as an action of the service handler.

I can remove this addition if you prefer, it's not even related to the matter at hand : the removal of Extended*.

Reply:
I can understand the motivation, but let's skip this for now. The helper method is not much code and the need for it is not too widely spread.

Reply:
Ok, I've removed it from the commit and updated the nemo-qml-plugin-calendar PR accordingly.